agent/engine/docker_task_engine.go

```diff
@@ -119,7 +121,7 @@ func NewDockerTaskEngine(cfg *config.Config, client DockerClient, credentialsMan
 // ImagePullDeleteLock ensures that pulls and deletes do not run at the same time.
```

Comment:
Can you update this comment to reflect the conditional nature of serialized pulls?